daffodil-core/src/main/scala/org/apache/daffodil/dsom/ElementBase.scala

```diff
@@ -342,7 +343,12 @@ trait ElementBase
           // will work.
           //
           val str = defaultValueAsString
-          val value = primType.fromXMLString(str)
+          val value = try {
+            primType.fromXMLString(str)
+          } catch {
+            case ipd: InvalidPrimitiveDataException =>
+              SDE("The default value (%s) is not a valid %s: %s".format(str, primType, ipd.getMessage))
```

Comment:
This seems like it's going to have repetitive information. We can asume InvalidPrimiteDataException has the information about the string value and the type, so we don't need to duplicated it this SDE. The only extra informaion this SDE needs to provided is that the invalid thing is a default value. How about something like this:
```suggestion
              SDE("Invalid default value: %s".format(ipd.getMessage))
```